### ISSUE TITLE
`test-rules` fails due to `test-driver` seg fault on armv6, armv7, aarch64, ppc64, ppc64le

### ISSUE THREAD

**Issue by ggardet** [VirusTotal/yara#1344]
`test-rules` fails due to `test-driver` seg fault on armv6, armv7, aarch64, ppc64, ppc64le.
Log:
```
[  453s] ./build-aux/test-driver: line 112: 29868 Segmentation fault      (core dumped) "$@" > $log_file 2>&1
[  453s] FAIL: test-rules
[  453s] make[3]: Leaving directory '/home/abuild/rpmbuild/BUILD/yara-4.0.2'
[  453s] make[3]: Entering directory '/home/abuild/rpmbuild/BUILD/yara-4.0.2'
[  453s] ==================================
[  453s]    yara 4.0.2: ./test-suite.log
[  453s] ==================================
[  453s] 
[  453s] # TOTAL: 13
[  453s] # PASS:  12
[  453s] # SKIP:  0
[  453s] # XFAIL: 0
[  453s] # FAIL:  1
[  453s] # XPASS: 0
[  453s] # ERROR: 0
[  453s] 
[  453s] .. contents:: :depth: 2
[  453s] 
[  453s] FAIL: test-rules
[  453s] ================
[  453s] 
[  453s] FAIL test-rules (exit status: 139)
[  453s] 
[  453s] ============================================================================
[  453s] Testsuite summary for yara 4.0.2
[  453s] ============================================================================
[  453s] # TOTAL: 13
[  453s] # PASS:  12
[  453s] # SKIP:  0
[  453s] # XFAIL: 0
[  453s] # FAIL:  1
[  453s] # XPASS: 0
[  453s] # ERROR: 0
[  453s] ============================================================================
```

**Comment by wxsBSD** [comment]
You wouldn't happen to have the core with debugging information available, would you?

**Comment by ggardet** [comment]
> You wouldn't happen to have the core with debugging information available, would you?

Unfortunately, no.

**Comment by wxsBSD** [comment]
I've got a Raspberry Pi 4 which I will see if I can replicate this on in the coming weeks. It may take me a bit because it's a brand new board and I have to get something running on it first. ;)

**Comment by wxsBSD** [comment]
I've managed to get a backtrace and this is related to #503 and actually comes from the fix in #586 - I'll keep digging...

**Comment by ggardet** [comment]
I confirm that https://github.com/VirusTotal/yara/pull/1352 fixes the segfault. Thanks!
